### ISSUE TITLE
The `Makefile` should use the version of `fix-agda-whitespace` in the repository

### ISSUE THREAD

**Issue by asr** [agda/agda#2318]
Currently, the `Makefile` use the version installed in the path.

**Comment by andreasabel** [comment]
On master, I now get:
```
...
Installed Agda-2.6.0
src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace --check
Whitespace violation detected in /home/abel/agda-master/dist-2.4.2.3/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.4.3/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.4.3-ghc-7.10/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.5/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.5.0/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.5.1.1/build/Agda/Syntax/Parser/Parser.hs.
Whitespace violation detected in /home/abel/agda-master/dist-2.6.0/build/Agda/Syntax/Parser/Parser.hs.
Makefile:332: recipe for target 'check-whitespace' failed
make: *** [check-whitespace] Error 1
```
If I run the installed fix-agda-whitespace, everything checks fine:
```
abel@agda2:~/agda-master$ fix-agda-whitespace --check
abel@agda2:~/agda-master$ 
```
Also, to fix the whitespace violations, I have now to run the lengthy
```
src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace
```

Would it not make sense to ignore the generated `Parser.hs` in all `dist` directories?

**Comment by asr** [comment]
> Would it not make sense to ignore the generated `Parser.hs` in all `dist` directories?

After 765cbb9 `fix-agda-withespace` excludes the `dist*` directories. I don't know why are you seeing whitespace violations on those directories.

> Also, to fix the whitespace violations, I have now to run the lengthy
>```
>src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace
>```

You can run `make fix-whitespace`.

 I added a missing `cabal clean`. After pulling, could you reproduce the issue?

**Comment by andreasabel** [comment]
For `make test` I now get:
```
src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace --check
bash: src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace: No such file or directory
Makefile:333: recipe for target 'check-whitespace' failed
make: *** [check-whitespace] Error 127
```
I think the goal `fix-whitespace` should trigger the goal `build-fix-agda-whitespace` if `src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace` is not present.

**Comment by asr** [comment]
> I think the goal `fix-whitespace` should trigger the goal `build-fix-agda-whitespace` if `src/fix-agda-whitespace/dist/build/fix-agda-whitespace/fix-agda-whitespace` is not present.

Testing for the binary is not enough because it can be obsolete, so I added as prerequisite `build-fix-agda-whitespace`. 

I had added `cabal clean` because if someone writes something like

```
make build-fix-agda-whitespace
change-ghc-version
make build-fix-agda-whitespace
```
she gets an error like:

```
Building fix-agda-whitespace-0.0.4...
Preprocessing executable 'fix-agda-whitespace' for
fix-agda-whitespace-0.0.4...
<command line>: cannot satisfy -package-id base-4.9.0.0
    (use -v for more information)
make: *** [build-fix-agda-whitespace] Error 1
```

Since we use `cabal clean && cabal build` it means that every time we run `make fix-whitespace` or `make check-whitespace` the program is recompiled. This isn't a nice solution, but it should avoid all the possible issues.

**Comment by andreasabel** [comment]
It seems reasonably fast, so, I don't think it is a problem it is recompiled each time.
